src/main/java/edu/project/yourpackage/WeatherApi.java

```diff
@@ -0,0 +1,47 @@
+//API key: 23c541de6a2105ebf03f39791a6b7cd1
+package edu.project.yourpackage;
+
+import com.google.gson.Gson;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+public class WeatherApi {
+    private static final String API_KEY = "23c541de6a2105ebf03f39791a6b7cd1";
```

Comment:
Let's move this `API_KEY` to the `Configuration` class as member `OPENWEATHERMAP_API_KEY`